homeassistant/components/recorder/db_schema.py

```diff
@@ -572,10 +571,12 @@ def shared_attrs_bytes_from_event(
         exclude_attrs = set(ALL_DOMAIN_EXCLUDE_ATTRS)
         if base_platform_attrs := exclude_attrs_by_domain.get(domain):
             exclude_attrs |= base_platform_attrs
-        if (entity_info := entity_sources.get(state.entity_id)) and (
-            integration_attrs := exclude_attrs_by_domain.get(entity_info["domain"])
-        ):
-            exclude_attrs |= integration_attrs
+        entity_info: EntityInfo | None = event.data.get("entity_info")
+        if entity_info:
+            if entity_attrs := entity_info["recorder_excluded_attributes"]:
+                exclude_attrs |= entity_attrs
+            if integration_attrs := exclude_attrs_by_domain.get(entity_info["domain"]):
+                exclude_attrs |= integration_attrs
```

Comment:
This too can be removed when all integrations with recorder platforms have been migrated